### PR TITLE
SAS-126 # added necessary functionality to gameAPI for add game

### PR DESCRIPTION
**Pull request by MrBahur**

### CODE REVIEW

File: src/main/java/com/SAS/soccer_association_system/GamesAPIController.java [hunk 1]
```diff
@@ -37,6 +37,11 @@ public String getGameEvents(@PathVariable String gameID) {
         return GameCRUD.getGameEvents(gameID).toString();
     }
 
+    @GetMapping(value = "/referee/{gameID}")
+    public String getRefereeForGame(@PathVariable String gameID) {
+        return GameCRUD.getRefereesName(gameID).toString();
```
Comment (yaarm) [inline src/main/java/com/SAS/soccer_association_system/GamesAPIController.java]: It suppose to call a function in the sasApplication and from there call the gameController and from there to the GameCRUD.

File: src/main/java/com/SAS/soccer_association_system/UserAPIController.java [hunk 2]
```diff
@@ -21,25 +22,36 @@
...
         JSONObject json = new JSONObject(credentials);
         String username = json.get("username").toString();
         String password = json.get("password").toString();
         String clientURL = request.getRemoteAddr() + ":" + request.getRemotePort();
         String role = app.login(username, password, clientURL);
-        return role.isEmpty() ? "Failed" : role;
+        if (!role.isEmpty()) {
+            JSONObject toReturn = new JSONObject();
+            toReturn.put("role", role);
+            String userId = UsersCRUD.getUserIdByUserName(username) + "";
```
Comment (yaarm) [inline src/main/java/com/SAS/soccer_association_system/UserAPIController.java:38]: Same here, you can just change the login function - app.login, that will return List of string or maybe a pair.

**Review by yaarm — CHANGES_REQUESTED** [review]

File: src/main/java/com/SAS/soccer_association_system/UserAPIController.java (hunk 2 above)
Comment (MrBahur) [reply src/main/java/com/SAS/soccer_association_system/UserAPIController.java:38]: working with JSON is easyer

File: src/main/java/com/SAS/soccer_association_system/GamesAPIController.java (hunk 1 above)
Comment (MrBahur) [reply src/main/java/com/SAS/soccer_association_system/GamesAPIController.java]: added a TBD

**Review by dekellevy93 — APPROVED** [review]

**Pull request merged**